pkg/controller/constraint/constraint_controller.go

```diff
@@ -356,11 +356,36 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 			if err != nil {
 				return reconcile.Result{}, err
 			}
-			vapBindingName := getVAPBindingName(instance.GetName())
-			currentVapBinding.SetName(vapBindingName)
-			if err := r.writer.Delete(ctx, currentVapBinding); err != nil {
-				if !apierrors.IsNotFound(err) {
-					return reconcile.Result{}, err
+			ct := &v1beta1.ConstraintTemplate{}
+			err = r.reader.Get(ctx, types.NamespacedName{Name: strings.ToLower(instance.GetKind())}, ct)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					return reconcile.Result{}, nil
+				}
+				return reconcile.Result{}, err
+			}
+			unversionedCT := &templates.ConstraintTemplate{}
+			if err := r.scheme.Convert(ct, unversionedCT, nil); err != nil {
+				return reconcile.Result{}, err
+			}
+			hasVAP, err := ShouldGenerateVAP(unversionedCT)
+			if err != nil {
+				if errors.Is(err, celSchema.ErrCELEngineMissing) {
+					return reconcile.Result{}, nil
```

Comment:
do we want to add a log before return

Reply:
I was reluctant to add a log statement here because with VAP beta this will be logged for all CT.